### ISSUE TITLE
'let' should not be allowed to be bound as a variable in a for-of loop or let/const decl

### ISSUE THREAD

**Issue by DanielRosenwasser** [Microsoft/TypeScript#3475]
The following examples should trigger an early error in JS:

```TypeScript
let let = 10;
```

```TypeScript
const let = 10;
```

<sub>Note: a similar test exists in `tests/cases/conformance/es6/for-ofStatements/for-of56.ts`) for the following 3 examples</sub>
```TypeScript
for (var let of [1,2,3]) {
}
```

```TypeScript
for (let let of [1,2,3]) {
}
```

```TypeScript
for (const let of [1,2,3]) {
}
```

I'm not sure if the following should trigger an error:

```TypeScript
var let = 10;
for (let of [1,2,3]) {
}
```

-------
-------

> ### <span id="sec-13.2.1.1" class="secnum">[13.2.1.1](https://people.mozilla.org/~jorendorff/es6-draft.html#sec-let-and-const-declarations-static-semantics-early-errors "link to this section")</span> Static Semantics: Early Errors

> *LexicalDeclaration* <span class="geq">:</span> *LetOrConst* *BindingList* `;`

> -   It is a Syntax Error if the BoundNames of *BindingList* contains `"let"`.

-----------------

> ### <span id="sec-13.6.4.1" class="secnum">[13.6.4.1](https://people.mozilla.org/~jorendorff/es6-draft.html#sec-for-in-and-for-of-statements-static-semantics-early-errors "link to this section")</span> Static Semantics: Early Errors

> *IterationStatement* <span class="geq">:</span>

> `for` `(` *ForDeclaration* `in` *Expression* `)` *Statement*

> `for` `(` *ForDeclaration* `of` *AssignmentExpression* `)` *Statement*

> -   It is a Syntax Error if the BoundNames of *ForDeclaration* contains `"let"`.

**Comment by yuit** [comment]
```TypeScript
var let = 10;
for (let of [1,2,3]) {
}
```
This should trigger an early error from this grammar production
```
for ( [lookahead ≠ let ] LeftHandSideExpression[?Yield] of AssignmentExpression[In, ?Yield] ) 
```

**Comment by DanielRosenwasser** [comment]
Yup - it would be useful to have a test like

```TypeScript
var let: number[] = []
let i = false
let s = 0
for (let[i] of [[1],[2],[3],[4],[5]]) {
    s += i
}
```

As well as one like this:

```TypeScript
var let: number[] = []
let i = 0
let s = 0
for ((let[i]) of [1,2,3,4,5]) {
    s += let[i]
    i++
}
```

Neither of these should error.

**Comment by DanielRosenwasser** [comment]
:100:
